drivers/bluetooth/hci/h4.c

```diff
@@ -344,9 +344,8 @@ static inline void read_payload(void)
 	    (evt_flags & BT_HCI_EVT_FLAG_RECV_PRIO)) {
 		LOG_DBG("Calling bt_recv_prio(%p)", buf);
 		bt_recv_prio(buf);
-	}
-
-	if (evt_flags & BT_HCI_EVT_FLAG_RECV) {
+	} else if ((evt_flags & BT_HCI_EVT_FLAG_RECV) ||
+		   (evt_flags & BT_HCI_EVT_FLAG_RECV_PRIO)) {
```

Comment:
That's a good catch. However, the only "receive" flags an HCI packet can have are either "normal" or "priority", so I don't think you need to explicitly test for these. Instead you can just extend the first if-statement with basic `} else {` branch.

Reply:
Indeed, this was my first choice, but I have checked the function  bt_hci_evt_get_flags and it seems that there are cases when both flags are set.

Reply:
Both flags being set means that the host will process the event in both priority and normal context. It doesn't really make any difference here though - my point was that it doesn't matter what flags (or how many) are set, you still want to just call `bt_recv()` in the new else-branch.

Reply:
Correction, call `net_buf_put()` in the else branch (the actual `bt_recv()` happens from another thread)

Reply:
Thanks for clarification, I will make the changes.
I still have a doubt, with my proposal or with your simplified version, there is a change in behavior in the following condition
- `CONFIG_BT_RECV_BLOCKING` set
- `BT_HCI_EVT_FLAG_RECV_PRIO` and `BT_HCI_EVT_FLAG_RECV` both set

Current version is calling both `bt_recv_prio` and `bt_recv` (via `net_buf_put`), while new version is only calling `bt_recv_prio`. Is this correct ?